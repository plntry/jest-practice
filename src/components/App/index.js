import Info from "../Info";

function App() {
    return (
      <div className="App">
        <h1>Hello World!</h1>
        <Info user="plntry" />
        <Info user="mplesha" />
      </div>
    );
  }
  
  export default App;
  