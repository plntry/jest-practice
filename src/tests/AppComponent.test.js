/* eslint-disable testing-library/await-async-query */
import renderer from "react-test-renderer";
import App from "../components/App";

describe("tests for App Component", () => {
  it("should render itself correctly", () => {
    const wrapper = renderer.create(<App />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it("should render header with text", () => {
    const wrapper = renderer.create(<App />);
    const testInstance = wrapper.root;
    expect(testInstance.findByProps({ children: 'Hello World!' })).toBeTruthy();
  });

  it("should render Info Component twice", () => {
    const wrapper = renderer.create(<App />);
    const testInstance = wrapper.root;
    expect(testInstance.findAllByProps({ children: 'GitHub User Info' }).length).toEqual(2);
  });
});
