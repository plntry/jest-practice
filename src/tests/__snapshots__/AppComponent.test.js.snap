// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for App Component should render itself correctly 1`] = `
<div
  className="App"
>
  <h1>
    Hello World!
  </h1>
  <div>
    <h3>
      GitHub User Info
    </h3>
    <ul />
  </div>
  <div>
    <h3>
      GitHub User Info
    </h3>
    <ul />
  </div>
</div>
`;
